test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Formatter/Serialization/ODataResourceSerializerTests.cs

```diff
@@ -892,6 +902,280 @@ public void CreateResource_Works_ToAppendDynamicProperties_ForOpenEntityType()
             Assert.Equal(new DateTimeOffset(dateTime), dateTimeProperty.Value);
         }
 
+        [Fact]
+        public void CreateResource_Works_WithInstanceAnnotations()
+        {
+            // Arrange
+            IEdmModel model = SerializationTestsHelpers.SimpleOpenTypeModel();
+
+            IEdmEntitySet customers = model.EntityContainer.FindEntitySet("Customers");
+
+            IEdmEntityType customerType = model.FindDeclaredType("Default.Customer") as IEdmEntityType;
+            Type simpleOpenCustomer = typeof(SimpleOpenCustomer);
+            model.SetAnnotationValue(customerType, new ClrTypeAnnotation(simpleOpenCustomer));
+
+            IEdmComplexType addressType = model.FindDeclaredType("Default.Address") as IEdmComplexType;
+            Type simpleOpenAddress = typeof(SimpleOpenAddress);
+            model.SetAnnotationValue(addressType, new ClrTypeAnnotation(simpleOpenAddress));
+
+            IEdmEnumType enumType = model.FindDeclaredType("Default.SimpleEnum") as IEdmEnumType;
+            Type simpleEnumType = typeof(SimpleEnum);
+            model.SetAnnotationValue(enumType, new ClrTypeAnnotation(simpleEnumType));
+
+            model.SetAnnotationValue(customerType, new InstanceAnnotationDictionaryAnnotation(
+            simpleOpenCustomer.GetProperty("InstanceAnnotations")));
+
+            var in1 = new Dictionary<string, object>();
+            in1.Add("NS.test1", 123);
+
+            var in2 = new Dictionary<string, object>();
+            in2.Add("NS.test2", 345);
+
+            var instAnn = new Dictionary<string, IDictionary<string, object>>();
+            instAnn.Add("", in1);
```

Comment:
>"" [](start = 24, length = 2)

empty for the resource or "null" for the resource?

Reply:
@ sreejith - Please confirm that I can pass null to add an instance annotation to the resource.

---
In reply to: [450395070](https://github.com/OData/WebApi/pull/2219#discussion_r450395070) [](ancestors = 450395070)

Reply:
No longer applicable with the new interface.

---
In reply to: [487242796](https://github.com/OData/WebApi/pull/2219#discussion_r487242796) [](ancestors = 487242796,450395070)